src/blocks/battery.rs

```diff
@@ -221,34 +228,94 @@ impl BatteryConfig {
     fn default_show() -> ShowType {
         ShowType::Percentage
     }
+
+    fn default_removable() -> bool {
+        false
+    }
 }
 
 impl ConfigBlock for Battery {
     type Config = BatteryConfig;
 
     fn new(block_config: Self::Config, config: Config, _tx_update_request: Sender<Task>) -> Result<Self> {
+        // If the device does not exist and its not removable, throw an Error
+        // If it is removable, set to None or Some(dev) depending on its status
+        let device =
+            match (
+                block_config.removable,
+                PowerSupplyDevice::from_device(block_config.device.clone())
+            ) {
+                (_, Ok(dev)) => Some(dev),
+                (true, Err(_)) => None,
+                (false, Err(e)) => return Err(e),
+            };
+
         Ok(Battery {
             id: Uuid::new_v4().simple().to_string(),
             update_interval: block_config.interval,
             output: TextWidget::new(config),
-            device: try!(PowerSupplyDevice::from_device(block_config.device)),
+            device_name: block_config.device,
+            device,
             show: block_config.show,
+            removable: block_config.removable,
         })
     }
 }
 
+// TODO: Is there a builtin for this?
```

Comment:
I'm almost certain there is something in the standard library for this. Can't find it however.